### PR TITLE
handle messagetype being null

### PR DESCRIPTION
**Pull request by PhilBastian**
fixes the symptoms of #3980 

handle messageType being null when processing an endpoint. Note that any null endpoints get filtered out of the displayed results set, in [messageTypes.ts](https://github.com/Particular/ServicePulse/blob/5af9e3a68b66b9a7ba750cd937c0af42353b97e4/src/Frontend/src/components/monitoring/messageTypes.ts#L54):

```js
      // filter out system message types
      .filter((mt) => mt.id && mt.typeName)
```

There is another discussion to be held as to whether unknown and uninferrable messagetypes should be passed through as null, since this prevents metrics being shown for these messages, but this is an imperfect first step to prevent an exception when handling this case

### CODE REVIEW

File: src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
     {
         public EndpointMessageType(string endpointName, string enclosedMessageTypes)
         {
-            var index = enclosedMessageTypes.IndexOf(';');
+            var index = (enclosedMessageTypes ?? "").IndexOf(';');
 
             var firstType = index != -1
```
Comment (johnsimons) [inline src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs:9]: I assume it is ok to set `MessageType` to `null`?

**Review by johnsimons — APPROVED** [review]
A small comment regarding what are the implications of having `MessageType` set top `null`, but otherwise it makes sense

File: src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs (hunk 1 above)
Comment (PhilBastian) [reply src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs:9]: it is handled fine by ServicePulse, which filters out such message types

Comment (PhilBastian) [reply src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs:9]: added filter servicecontrol side (any metrics gathered against a null messagetype will be discarded anyway)

**Comment by johnsimons** [comment]
@PhilBastian i thought we were going to see if we could filter out these messages at deserialisation time?

**Pull request merged**